carp.studies.core/src/commonMain/kotlin/dk/cachet/carp/studies/domain/users/ParticipantRepository.kt

```diff
@@ -20,6 +20,11 @@ interface ParticipantRepository
      */
     suspend fun getRecruitment( studyId: UUID ): Recruitment?
 
+    /**
+     * Returns all stored [Recruitment]s.
+     */
+    suspend fun getRecruitments(): List<Recruitment>
```

Comment:
This is a very bad idea. It's a full table dump and will inevitably get slower and slower.

I guess you added this to be able to drop the `studyId` parameter. But, the way the repository pattern in DDD works is that you are supposed to expose the _queries_ you need from the domain layer in the interface (the use cases), instead of the very typical simplistic CRUD operations and end up writing query code in the layer above.

So, instead what we likely need is a `getRecruitmentWithParticipantGroup( groupId )`. This can be optimized by whatever underlying DB technology and doesn't pull in unnecessary data.